### ISSUE TITLE
Add download_file to Script Runner

### ISSUE THREAD

**Issue by GTI-Dakota** [OpenC3/cosmos#645]
Hello,

Is there a way to grab a file in the script runner and then save it to a local machine like the data extractor does? I noticed that COSMOS 5 deprecated the save_file_dialog, so I was wondering if there is another way to do this. The flow would be as follows:

- Script starts
- Create File
- Save file to minio
- Open Save File Dialog so user can save to local machine
- Script Ends

Thanks,

Dakota

**Comment by jmthomas** [comment]
You can use [put_target_file](https://openc3.com/docs/v5/scripting#put_target_file) to write to Buckets.  Also see [target_file.rb](https://github.com/OpenC3/cosmos/blob/main/openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST/procedures/target_file.rb) from the DEMO.

**Comment by GTI-Dakota** [comment]
Well, I'm not looking to save to minio, I'm looking to pull from minio to save to my machine via a script

**Comment by GTI-Dakota** [comment]
Is there a way to leverage the same API that the data extractor uses to allow a user to save a file from minio? Like in the bucket explorer with the download button. Is that capability able to be called via script?

**Comment by jmthomas** [comment]
We have the `open_file_dialog` and `open_files_dialog` but don't yet have the `save_file_dialog` method. We can add this in an upcoming release. Right now the work around would be to tell the user that you saved it to a specific location and then open Bucket Explorer to that location so they can download it.

**Comment by GTI-Dakota** [comment]
>Right now the work around would be to tell the user that you saved it to a specific location and then open Bucket Explorer to that location so they can download it.

We do that now! It is pretty helpful as is but we want that final leap to get the file to our computer after a massive suite has been ran, so we can easily save it to our shared drives. Thanks for adding it in the next release!

**Comment by timothyscherer** [comment]
I was using this in COSMOS 5 and recently upgraded to 6.8.1 where it is no longer present. Is there a plan to add support back, or an alternative approach to get the same solution? I really appreciated this feature.

EDIT: Maybe this is actually a bug, the method is present but it has an attribute error:
`AttributeError: module 'openc3.script' has no attribute '_get_download_url'`

From `running_script.py`:
```py
def download_file(path, scope=OPENC3_SCOPE):
    url = openc3.script._get_download_url(path, scope=scope)
    running_script_anycable_publish(
        f"running-script-channel:{RunningScript.instance.id()}",
        {"type": "downloadfile", "filename": os.path.basename(path), "url": url},
    )


setattr(openc3.script, "download_file", download_file)
```

**Comment by ryanmelt** [comment]
Its a bug.  Still working in Ruby but looks like Python is broken.

**Comment by timothyscherer** [comment]
Should I submit a new issue or is it covered? Any chance a fix for this works its way into 6.8.2?

**Comment by ryanmelt** [comment]
I created an issue.  We'll get it fixed for 6.9.0 this week.
